tests/index.spec.ts

```diff
@@ -0,0 +1,265 @@
+        },
+        'Can parse URI with payload starting with `0x`'
+    );
+
+    t.deepEqual(
+        parse('ethereum:pay-0x1234DEADBEEF5678ABCD1234DEADBEEF5678ABCD'),
+        {
+            scheme: 'ethereum',
+            prefix: 'pay',
+            target_address: '0x1234DEADBEEF5678ABCD1234DEADBEEF5678ABCD',
+        },
+        'Can parse URI with payload starting with `0x` and `pay` prefix'
+    );
+
+    t.deepEqual(
+        parse('ethereum:foo-0x1234DEADBEEF5678ABCD1234DEADBEEF5678ABCD'),
+        {
+            scheme: 'ethereum',
+            prefix: 'foo',
+            target_address: '0x1234DEADBEEF5678ABCD1234DEADBEEF5678ABCD',
+        },
+        'Can parse URI with payload starting with `0x` and `foo` prefix'
+    );
+
+    t.deepEqual(
+        parse('ethereum:foo-doge-to-the-moon.eth'),
+        {
+            scheme: 'ethereum',
+            prefix: 'foo',
+            target_address: 'doge-to-the-moon.eth',
+        },
+        'Can parse URI with an ENS name'
+    );
+
+    t.deepEqual(
+        parse('ethereum:0x1234DEADBEEF5678ABCD1234DEADBEEF5678ABCD@42'),
+        {
+            scheme: 'ethereum',
+            target_address: '0x1234DEADBEEF5678ABCD1234DEADBEEF5678ABCD',
+            chain_id: '42',
+        },
+        'Can parse URI with chain id'
+    );
+
+    t.deepEqual(
+        parse(
+            'ethereum:0x1234DEADBEEF5678ABCD1234DEADBEEF5678ABCD/transfer?address=0x12345&uint256=1'
+        ),
+        {
+            scheme: 'ethereum',
+            target_address: '0x1234DEADBEEF5678ABCD1234DEADBEEF5678ABCD',
+            function_name: 'transfer',
+            args: [
+                ['address', '0x12345'],
+                ['uint256', '1'],
+            ],
+        },
+        'Can parse an ERC20 token transfer'
+    );
+
+    t.deepEqual(
+        parse(
+            'ethereum:0x1234DEADBEEF5678ABCD1234DEADBEEF5678ABCD?value=2.014e18&gas=10&gasLimit=21000&gasPrice=50'
+        ),
+        {
+            scheme: 'ethereum',
+            target_address: '0x1234DEADBEEF5678ABCD1234DEADBEEF5678ABCD',
+            parameters: {
+                value: '2014000000000000000',
+                gas: '10',
+                gasLimit: '21000',
+                gasPrice: '50',
+            },
+        },
+        'Can parse a url with value and gas parameters'
+    );
+
+    t.deepEqual(
+        parse(
+            'ethereum:0x1234DEADBEEF5678ABCD1234DEADBEEF5678ABCD?value=1&gas=10&gasLimit=21000&gasPrice=50&uint256=-2.014e18'
+        ),
+        {
+            scheme: 'ethereum',
+            target_address: '0x1234DEADBEEF5678ABCD1234DEADBEEF5678ABCD',
+            parameters: {
+                value: '1',
+                gas: '10',
+                gasLimit: '21000',
+                gasPrice: '50',
+            },
+            args: [['uint256', '-2014000000000000000']],
+        },
+        'Can parse a url with negative value and gas parameters'
```

Comment:
I think in this test it should throw  "missing function_name"
"args" require a`function_name` to be present, otherwise they are pointless.